src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -281,19 +283,64 @@ private static boolean isDate(String token) {
   }
 
   /**
-   * Checks the raw text in the builder for prices that can be parsed. The largest price found, if
-   * it exists, is added to the builder.
+   * Searches the raw text in the builder for the total price using multiple heuristics. If any find
+   * a price, it is added to the builder.
    */
   private static void checkForParsablePrices(AnalysisResults.Builder analysisBuilder) {
-    // Assume that the largest price on the receipt is the total price.
+    if (!findPriceAfterTotal(analysisBuilder)) {
+      findLargestPrice(analysisBuilder);
+    }
+  }
+
+  /**
+   * Checks the raw text in the builder for a price appearing after the word "total" and adds it to
+   * the builder if found. The return value indicates whether a price was added.
```

Comment:
Use a java tag here instead of a sentence. So it would be "@return whether a price was added." 

https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#format